charlotte_core/src/arch/x86_64/memory/pmm.rs

```diff
@@ -32,11 +32,20 @@ lazy_static! {
     ///The physical frame allocator to be used by the kernel and user-space applications.
     pub static ref PFA: TicketMutex<PhysicalFrameAllocator> = TicketMutex::new(PhysicalFrameAllocator::new());
 }
+/// The number of frames that may need to be allocated but have not been allocated yet.
+/// To be used for things like faulted in pages and copy-on-write pages.
+/// This value is used to determine how overcommitted the system is.
+/// When the system is overcommitted by more than a certain percentage, new allocations will fail.
+static mut UNALLOCATED_FRAMES_COMMITTED: TicketMutex<usize> = TicketMutex::new(0);
```

Comment:
Does this need to be a `static mut`, we should probably be avoiding these when possible.

Reply:
Yes it will be added to everytime the VMM creates a faulted in region or CoW region in any virtual address space since the PFA should track how many unallocated pages have been committed to by the system.